src/review/review.service.ts

```diff
@@ -29,97 +31,116 @@ export class ReviewService {
   ) {}
 
   async createReview(reviewDetails: CreateReviewDto) {
-    const { hotLevel, userId, foodId, tags } = reviewDetails;
-    const review = new Review();
-    const hotLevelId = produceHotLevelId(hotLevel);
-    review.hotLevel = await getRepository(FoodLevel).findOne(hotLevelId);
-    review.user = await getRepository(User).findOne(userId);
-    review.food = await getRepository(Food).findOne(foodId);
-    review.tasteReviews = await getRepository(TasteTag).find({
-      name: In(tags),
-    });
-
-    const result = await this.reviewRepository.save(review);
-
-    return {
-      userId: result.user.id,
-      foodId: result.food.id,
-    };
-  }
-
-  async createReviews(reviewsDetails: CreateReviewsDto) {
-    const { userId, reviewList } = reviewsDetails;
-
-    reviewList.map(async ({ foodId, hotLevel, tags }) => {
+    try{
+      const { hotLevel, userId, foodId, tags } = reviewDetails;
       const review = new Review();
       const hotLevelId = produceHotLevelId(hotLevel);
-
+      review.hotLevel = await getRepository(FoodLevel).findOne(hotLevelId);
       review.user = await getRepository(User).findOne(userId);
       review.food = await getRepository(Food).findOne(foodId);
-      review.hotLevel = await getRepository(FoodLevel).findOne(hotLevelId);
-      review.tasteReviews = await getRepository(TasteTag).find({
-        name: In(tags),
+      review.tasteReviews = await Promise.all(
+        tags.map(async (tagid) => {
+          const tag = getRepository(TasteTag).findOne(tagid);
+          return tag;
+        }),
+      );
+      const result = await this.reviewRepository.save(review);
+      return {
+        userId: result.user.id,
+        foodId: result.food.id,
+      };
+    } catch (e) {
+      throw new HttpException(ERROR_MESSAGE.NOT_FOUND, HttpStatus.BAD_REQUEST);
+    }
+  }
+
+  async createReviews(reviewsDetails: CreateReviewsDto) {
+    try{
+      const { userId, reviewList } = reviewsDetails;
+
+      reviewList.map(async ({ foodId, hotLevel, tags }) => {
+        const review = new Review();
+        const hotLevelId = produceHotLevelId(hotLevel);
+
+        review.user = await getRepository(User).findOne(userId);
+        review.food = await getRepository(Food).findOne(foodId);
+        review.hotLevel = await getRepository(FoodLevel).findOne(hotLevelId);
+        review.tasteReviews = await Promise.all(
+          tags.map(async (tagid) => {
+            const tag = getRepository(TasteTag).findOne(tagid);
+            return tag;
+          }),
+        );
+        return this.reviewRepository.save(review);
       });
-      
-      return this.reviewRepository.save(review);
-    });
 
-    return {
-      userId: userId,
-      reviewLength: reviewList.length,
-    };
+      return {
+        userId: userId,
+        reviewLength: reviewList.length,
+      };
+    } catch (e) {
+      throw new HttpException(ERROR_MESSAGE.NOT_FOUND, HttpStatus.BAD_REQUEST);
```

Comment:
```suggestion
      throw new HttpException(ERROR_MESSAGE.BAD_REQUEST, HttpStatus.BAD_REQUEST);
```
해당 부분도 위와 같이 변경하는게 좋을 것 같습니다!